lib/dependency.js

```diff
@@ -45,16 +48,60 @@ Dependency.from = {
     return result;
   },
   csprojEntry: function (referenceItem) {
-    debug('Extracting by hintPath ' + referenceItem.HintPath[0]);
+    var hintPath = referenceItem.HintPath[0];
+    var result;
+    var version;
+    var name;
+
+    if (isWindows) {
+      try {
+        // try execute powershell to extract version from .dll file
+        var command = '(Get-Item ${file}).VersionInfo.ProductVersion'
+          .split('${file}').join(hintPath);
+        version = childProcess.execSync('powershell.exe ' + command, {
```

Comment:
We use `child_process` a bit differently in other plugins, see https://github.com/snyk/snyk-mvn-plugin/blob/b656bf0071b4b633991241459de8d8e786bb31a1/lib/sub-process.js#L13

Not sure what the difference is, but remember that `spawn` was preferred for one reason or another. Can you check, replace if it makes sense?

Reply:
spawn is used to spawn another node. execSync will execute the powershell in a synchronous manner to avoid mixed handling of failures and success. exec and spawn have different purposes.